fdtd/fdtd2d.py

```diff
@@ -70,6 +70,12 @@ def computeRHSH(self, fields):
         rhsH = - self.cEy*(Ex[1:, :] - Ex[:-1, :]) \
                + self.cEx*(Ey[:, 1:] - Ey[:, :-1])
 
+        if self.boundary_label == "PMC":
+            rhsH[0, :] = 0.0
+            rhsH[-1, :] = 0.0
+            rhsH[:,  0] = 0.0
+            rhsH[:, -1] = 0.0
+
```

Comment:
This is not correct. You are not setting H as zero in the boundary but in the previous inner cell. Please review MCFNL slides to check how to set PMC on boundaries.

Reply:
Start with the 1D case if you have doubts and to do tests quicker.